salt/modules/hosts.py

```diff
@@ -34,7 +32,7 @@ def _get_or_create_hostfile():
     if hfn is None:
         hfn = ''
     if not os.path.exists(hfn):
-        with salt.utils.files.fopen(hfn, 'w'):
+        with salt.utils.fopen(hfn, 'w'):
```

Comment:
Did you based this on recent code?  Strange to be changing `salt.utils.files.fopen` to `salt.utils.fopen`

Reply:
This had to be done due to the version we're using, so i'm assuming that the one i removed is the newer way of doing it right? I'll gladly change it back.

Reply:
@damon-atkins these limits are only present in Windows as far as we can tell, but we did test on multiple versions of Windows Server (2016 and 2012).

Applying these limits to *nix OSs will make the hosts file more readable tho.

I couldn't find any info on this on an RFC per se, but [articles like these litter the internet](https://www.uow.edu.au/~blane/netapp/ontap/nag/networking/concept/c_oc_netw_maintaining_host_file_limits.html) (whithout citing sources).
I did test that at least in Ubuntu Xenial, the 255 char limit is not enforced.

Reply:
NetApp says [Maximum line size is 1022 characters](https://library.netapp.com/ecmdocs/ECMP1368834/html/GUID-C6F3B6D1-232D-44BB-A76C-3304C19607A3.html). Also [Maximum number of aliases is 34](https://library.netapp.com/ecmdocs/ECMP1368834/html/GUID-C6F3B6D1-232D-44BB-A76C-3304C19607A3.html)

Reply:
NetApp was the 4th google result. I didn't see the link @jcrsilva referenced from uow.edu.au after 5 pages of google search for "hosts file line limit", so "litters the internet" seems a bit dubious.

Reply:
And there is an RFC referenced by the /etc/hosts man page, 952.